### ISSUE TITLE
Add `nocorpus` linter

### ISSUE THREAD

**Issue by sivchari** [golangci/golangci-lint#3515]
### Your feature request related to a problem? Please describe.

The nocorpus linter detects fuzzing tests that has no corpus. Fuzzing test requires corpus.

### Valid case

```go
func FuzzOKTest(f *testing.F) {
	f.Add("x=1&y=2")
	f.Fuzz(func(t *testing.T, queryStr string) {
		query, err := url.ParseQuery(queryStr)
		if err != nil {
			t.Skip()
		}
		queryStr2 := query.Encode()
		query2, err := url.ParseQuery(queryStr2)
		if err != nil {
			t.Fatalf("ParseQuery failed to decode a valid encoded query %s: %v", queryStr2, err)
		}
		if !reflect.DeepEqual(query, query2) {
			t.Errorf("ParseQuery gave different query after being encoded\nbefore: %v\nafter: %v", query, query2)
		}
	})
```

### Invalid Case

```go
func FuzzNGTest(f *testing.F) {
	f.Fuzz(func(t *testing.T, queryStr string) {
		query, err := url.ParseQuery(queryStr)
		if err != nil {
			t.Skip()
		}
		queryStr2 := query.Encode()
		query2, err := url.ParseQuery(queryStr2)
		if err != nil {
			t.Fatalf("ParseQuery failed to decode a valid encoded query %s: %v", queryStr2, err)
		}
		if !reflect.DeepEqual(query, query2) {
			t.Errorf("ParseQuery gave different query after being encoded\nbefore: %v\nafter: %v", query, query2)
		}
	})

}
```

### Describe the solution you'd like.

I created a [nocorpus](https://github.com/sivchari/nocorpus) linter. If this liter is accepted, I would work to release this (v1.0.0).

### Describe alternatives you've considered.

-

**Comment by ldez** [comment]
In order for a pull request adding a linter to be reviewed, the linter and the PR must follow some requirements.

### Pull Request Description

- [ ] It must have a link to the linter repository.
- [ ] It must provide a short description of the linter.

### Linter

- [ ] It must not be a duplicate of another linter or a rule of a linter. (the team will help to verify that)
- [ ] It must have a valid license (AGPL is not allowed) and the file must contain the required information by the license, ex: author, year, etc.
- [ ] The linter repository must have a CI and tests.
- [ ] It must use [`go/analysis`](https://golangci-lint.run/contributing/new-linters/).
- [ ] It must have a valid tag, ex: `v1.0.0`, `v0.1.0`.
- [ ] It must not contain `init()`.
- [ ] It must not contain `panic()`, `log.fatal()`, `os.exit()`, or similar.
- [ ] It must not have false positives/negatives. (the team will help to verify that)
- [ ] It must have tests inside golangci-lint.

### The Linter Tests Inside Golangci-lint

- [ ] They must have at least one std lib import.
- [ ] They must work with `T=<name of the linter test file>.go make test_linters`. (the team will help to verify that)

### `.golangci.reference.yml`

- [ ] The linter must be added to the list of available linters (alphabetical case-insensitive order).
    - `enable` and `disable` options
- [ ] If the linter has a configuration, the exhaustive configuration of the linter must be added (alphabetical case-insensitive order)
    - The values must be different from the default ones.
    - The default values must be defined in a comment.
    - The option must have a short description.

### Others Requirements

- [ ] The files (tests and linter) inside golangci-lint must have the same name as the linter.
- [ ] The `.golangci.yml` of golangci-lint itself must not be edited and the linter must not be added to this file.
- [ ] The linters must be sorted in the alphabetical order (case-insensitive) in the `Manager.GetAllSupportedLinterConfigs(...)` and `.golangci.reference.yml`.
- [ ] The load mode (`WithLoadMode(...)`):
    - if the linter doesn't use types: `goanalysis.LoadModeSyntax`
    - `goanalysis.LoadModeTypesInfo` required `WithLoadForGoAnalysis()` in the `Manager.GetAllSupportedLinterConfigs(...)`
- [ ] The version in `WithSince(...)` must be the next minor version (`v1.X.0`) of golangci-lint.

### Recommendations

- [ ] The linter should not use SSA. (currently, SSA does not support generics)
- [ ] The linter repository should have a readme and linting.
- [ ] The linter should be published as a binary. (useful to diagnose bug origins)

---

The golangci-lint team will edit this comment to check the boxes before and during the review.

**This checklist does not imply that we will accept the linter.**

**Comment by leonklingele** [comment]
Doesn't it make sense to enforce this in the Go fuzzer itself?

**Comment by Antonboom** [comment]
Should be covered by https://golangci-lint.run/usage/linters/#govet (`tests` checker):
https://github.com/golang/tools/blob/8234134986237e66077e84b6916430f907a922c4/go/analysis/passes/tests/tests.go#L99
